tests/components/zwave_js/test_update.py

```diff
@@ -252,14 +251,57 @@ async def test_update_entity_dead(
 
     event = Event(
         "alive",
-        data={"source": "node", "event": "alive", "nodeId": multisensor_6.node_id},
+        data={"source": "node", "event": "alive", "nodeId": zen_31.node_id},
     )
-    multisensor_6.receive_event(event)
+    zen_31.receive_event(event)
     await hass.async_block_till_done()
 
     # Now that the node is up we can check for updates
     assert len(client.async_send_command.call_args_list) > 0
 
     args = client.async_send_command.call_args_list[0][0][0]
     assert args["command"] == "controller.get_available_firmware_updates"
-    assert args["nodeId"] == multisensor_6.node_id
+    assert args["nodeId"] == zen_31.node_id
+
+
+async def test_update_entity_ha_not_running(
+    hass,
+    client,
+    zen_31,
+):
+    """Test update occurs after HA starts."""
+    await hass.async_stop()
+
+    entry = MockConfigEntry(domain="zwave_js", data={"url": "ws://test.org"})
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert len(client.async_send_command.call_args_list) == 0
+
+    await hass.async_start()
+
+    assert len(client.async_send_command.call_args_list) == 1
+    args = client.async_send_command.call_args_list[0][0][0]
+    assert args["command"] == "controller.get_available_firmware_updates"
+    assert args["nodeId"] == zen_31.node_id
+
+
+async def test_update_entity_failure(
+    hass,
+    client,
+    climate_radio_thermostat_ct100_plus_different_endpoints,
+    controller_node,
+    integration,
+    hass_ws_client,
+):
+    """Test update entity update failed."""
+    assert len(client.async_send_command.call_args_list) == 0
+    client.async_send_command.side_effect = FailedZWaveCommand("test", 260, "test")
+
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(days=1))
+    await hass.async_block_till_done()
+
+    state = hass.states.get(UPDATE_ENTITY)
```

Comment:
for some reason in this test, after the time change, `len(client.async_send_command.call_args_list) = 2` and both commands are for the same node. I can't figure out why this is but I am worried that this is related to some bug that I need to fix. Any ideas?

Reply:
One call happens when Home Assistant starts and another after a day when we move time forward, right?

Reply:
We reset the mock in the integration fixture. That's why at the top of the test the assertion that the list is empty passes

Reply:
Ok.

Reply:
I don't see any problem yet.

Reply:
I'm seeing this error if I assert False at the end of the test:
```
2022-09-06 11:08:32.659 ERROR    MainThread homeassistant.components.zwave_js:__init__.py:719 Unexpected exception: Listen wasn't canceled!
assert False
Traceback (most recent call last):
  File "/home/martin/dev/home-assistant/core/homeassistant/components/zwave_js/__init__.py", line 712, in client_listen
    await client.listen(driver_ready)
  File "/usr/lib/python3.9/unittest/mock.py", line 2164, in _execute_mock_call
    result = await effect(*args, **kwargs)
  File "/home/martin/dev/home-assistant/core/tests/components/zwave_js/conftest.py", line 588, in listen
    assert False, "Listen wasn't canceled!"
AssertionError: Listen wasn't canceled!
```

Reply:
I think I've fixed this.